standalone/portal-api/src/modules/company-information/services/supplier-information.service.ts

```diff
@@ -0,0 +1,160 @@
+import { BuyerCompanyEntity } from 'src/modules/company/models/buyer.company.entity';
+import { BuyerCompanyEntityRepository } from 'src/modules/company/repositories/buyer.company.repository';
+import { triggerError } from 'src/common/trigger.error';
+
+@Injectable()
+export class SupplierInformationService extends BaseService {
+  constructor(
+    @InjectRepository(QuickApplicationEntity)
+    private buyerApplicationRepository: BuyerQuickApplicationEntityRepository,
+
+    @InjectRepository(BuyerCompanyEntity)
+    private buyerCompanyRepository: BuyerCompanyEntityRepository,
+
+    private readonly mappingService: EntityMappingService,
+    private readonly supplierService: SupplierCompanyEntityService,
+    private readonly relationshipService: RelationshipService,
+  ) {
+    super();
+  }
+
+  public async getSupplierInformation(
+    applicationId: string,
+  ): Promise<SupplierInformationResponse> {
+    const application = await this.buyerApplicationRepository.findOne(
+      applicationId,
+      {
+        relations: ['supplierInformation'],
+      },
+    );
+
+    const savedSupplier = await this.supplierService.getSupplierEntity(
+      application.supplierInformation.id,
+    );
+
+    const savedRelationship = await this.relationshipService.getEntity(
+      application.company.id,
+      application.supplierInformation.id,
+    );
+
+    return this.mappingService.mapSupplierInformationForImporterApplication(
+      savedSupplier.id,
+      savedRelationship,
+    );
+  }
+
+  @Transactional({
+    isolationLevel: IsolationLevel.READ_COMMITTED,
+  })
+  public async createSupplierInformation(
+    data: CreateSupplierInformationRequest,
+    applicationId: string,
+  ): Promise<SupplierInformationResponse> {
+    this.setupTransactionHooks();
+
+    //TODO: check for application so that isolated supplier creation is prevented
+    const application = await this.buyerApplicationRepository.findOne(
+      applicationId,
+      {
+        relations: [
+          'supplierInformation',
+          'paymentTerms',
+          'orderDetails',
+          'security',
+          'transactionDocuments',
+        ],
+      },
+    );
+    if (application.supplierInformation) {
+      throw new HttpException(
+        'Supplier information already exists',
+        HttpStatus.BAD_REQUEST,
+      );
+    }
+
+    const selectedSupplier = await this.supplierService.findSupplierEntityById(
+      data.supplierId,
+    );
+
+    const savedRelationship = await this.relationshipService.createEntity(
+      data.relationshipWithSupplier,
+      application.company.id,
+      selectedSupplier,
+    );
+
+    selectedSupplier.relationshipWithBuyers = [
+      ...selectedSupplier.relationshipWithBuyers,
+      savedRelationship,
+    ];
+
+    selectedSupplier.applications = [
+      ...selectedSupplier.applications,
+      application,
+    ];
+
+    selectedSupplier.save();
```

Comment:
What are we saving here? Why are we not calling repositories for saving stuff?

Reply:
when a supplier information is created i am updating the supplier company entity with the new application and the new relationship hence to save these changes. Im refactoring this to using repository.